src/metabase/server/middleware/session.clj

```diff
@@ -382,14 +382,22 @@
   :type    :json
   :default nil
   :getter  (fn []
-             (let [value (setting/get-value-of-type :json :session-timeout)]
-               (if-let [exception (check-session-timeout value)]
-                 (println (str "WARNING: " (ex-message exception)))
+             (let [value     (setting/get-value-of-type :json :session-timeout)
+                   error-key (check-session-timeout value)]
+               (case error-key
+                 :amount-must-be-positive
+                 (log/warn (trs "Session timeout amount must be positive."))
+                 :amount-must-be-less-than-100-years
+                 (log/warn (trs "Session timeout must be less than 100 years."))
                  value)))
   :setter  (fn [new-value]
-             (if-let [exception (check-session-timeout new-value)]
-               (throw exception)
-               (setting/set-value-of-type! :json :session-timeout new-value))))
+             (let [error-key (check-session-timeout new-value)]
+               (case error-key
+                 :amount-must-be-positive
+                 (throw (ex-info (tru "Session timeout amount must be positive.") {:status-code 400}))
+                 :amount-must-be-less-than-100-years
+                 (throw (ex-info (tru "Session timeout must be less than 100 years.") {:status-code 400}))
+                 (setting/set-value-of-type! :json :session-timeout new-value)))))
```

Comment:
This is just a suggestion for future code but here's a different way to structure this code.  

```suggestion
             (when-let [error-key (check-session-timeout new-value)]
               (throw (ex-info (case error-key
                                 :amount-must-be-positive            (tru "Session timeout amount must be positive.")
                                 :amount-must-be-less-than-100-years (tru "Session timeout must be less than 100 years."))
                               {:status-code 400}))
             (setting/set-value-of-type! :json :session-timeout new-value))))
```

I think it has a few benefits

1. The `case` expression would error on an unhandled error type -- the way it's structured now, because `setting/set-value-of-type!` is the fall-thru condition, if you had a typo in one of the error keywords it would go unhandled and nothing would catch it
2. It pushes the `case` statement down to the lowest point it can go and means the `(throw (ex-info ... {:status-code 400}))` doesn't have to be repeated twice
3. The unexceptional normal code path is kept at the top level of the function. It's less cognitive load to read code that is nested less (not a big difference here but over time little differences do add up)